upgrade-2.x.sh

```diff
@@ -0,0 +1,204 @@
+            ;;
+    esac
+else
+    log ERROR "No target OS version specified."
+fi
+
+# Check host OS version
+case $VERSION in
+    2.*)
+        log "Host OS version \"$VERSION\" OK."
+        ;;
+    *)
+        log ERROR "Host OS version \"$VERSION\" not supported."
+        ;;
+esac
+
+# Check if we need to install some more extra tools
+if ! version_gt "$VERSION" "$preferred_hostos_version" &&
+    ! [ "$VERSION" == "$preferred_hostos_version" ]; then
+    log "Host OS version $VERSION is less than $preferred_hostos_version, installing tools..."
+    tools_path=/tmp/upgrade_tools
+    tools_binaries="tar"
+    mkdir -p $tools_path
+    export PATH=$tools_path:$PATH
+    case $binary_type in
+        arm)
+            download_uri=https://github.com/resin-os/resinhup/raw/onetotwo/upgrade-binaries/$binary_type
+            for binary in $tools_binaries; do
+                log "Installing $binary..."
+                curl -s -L -o $tools_path/$binary $download_uri/$binary
+                chmod 755 $tools_path/$binary
+            done
+            ;;
+        *)
+            log ERROR "Binary type $binary_type not supported."
+            ;;
+    esac
+fi
+
+# Find which partition is / and which we should write the update to
+root_part=$(findmnt -n --raw --evaluate --output=source /)
+case $root_part in
+    *p2)
+        root_dev=${root_part%p2}
+        update_part=${root_dev}p3
+        update_part_no=3
+        update_label=resin-rootB
+        ;;
+    *p3)
+        root_dev=${root_part%p3}
+        update_part=${root_dev}p2
+        update_part_no=2
+        update_label=resin-rootA
+        ;;
+    *)
+        log ERROR "Unknown root partition \"$root_part\"."
+esac
+
+# Stop docker containers
+log "Stopping all containers..."
+systemctl stop update-resin-supervisor.timer > /dev/null 2>&1
+systemctl stop resin-supervisor > /dev/null 2>&1
+docker stop $(docker ps -a -q) > /dev/null 2>&1 || true
+
+image=resin/resinos:${target_version}-${SLUG}
+fsarchive=/mnt/data/newos.tar.gz
+
+log "Getting new OS image..."
+# Create container for new version
+CONTAINER=$(docker create "$image" echo export)
+
+log "Exporting new OS image..."
+# Export container
+docker export "$CONTAINER" | gzip > "$fsarchive"
+
+# Remove container
+docker rm "$CONTAINER"
+
+log "Making new OS filesystem..."
+# Format alternate root partition
+mkfs.ext4 -F -L "$update_label" "$update_part"
+
+# Mount alternate root partition
+mkdir -p /tmp/updateroot
+mount "$update_part" /tmp/updateroot
+
+# Extract rootfs
+log "Extracting new rootfs..."
+cat >/tmp/root-exclude <<EOF
+quirks
+resin-boot
+EOF
+tar -x -X /tmp/root-exclude -C /tmp/updateroot -f $fsarchive
+
+# Extract quirks
+tar -x -C /tmp -f $fsarchive quirks
+cp -a /tmp/quirks/* /tmp/updateroot/
+rm -rf /tmp/quirks
+
+# XXX hack required until 2.0.7 when a quirk is added for this
+(cd /tmp/updateroot/etc; rm mtab; ln -s ../proc/self/mounts mtab)
```

Comment:
Should we only support updates to 2.0.7+? I don't see the benefit in updating to lower versions and keep this hack.